### ISSUE TITLE
How to enable clustering feature using custom view annotation on mapbox android?

### ISSUE THREAD

**Issue by alexviet** [mapbox/mapbox-maps-android#1940]
i have a trouble about enable clustering feature on mapbox v10 android. I have tried to reseach on homepage but didn't find any helpful way to resolve this. 
I used to create point annotation manager from AnnotationPlugin before then i can add annotationConfig into them but now i must create custom view instead. 

So i don't know how to add clustering for that.
Here is way to add custom annotation view into mapview.
`val point = fromLngLat(
    longitude,
    latitude
)
val view = mMapView.viewAnnotationManager.addViewAnnotation(
         R.layout.custom_layout,
         viewAnnotationOptions {
              geometry(point)
              allowOverlap(true)
         }
)`

Anyone help me please? Thanks

**Comment by ank27** [comment]
@alexviet if you are looking for clusteringOptions using PointAnnotations, maybe this example might help 
https://github.com/mapbox/mapbox-maps-android/blob/main/app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/PointAnnotationClusterActivity.kt#L62-L72

**Comment by alexviet** [comment]
@ank27 Yep i used it before, it worked but for old requirement. As i told that i must add custom view instead create point annotation. So i can't use this way :( Can you give me another way?

**Comment by ank27** [comment]
@alexviet Yes you can defenitely add ViewAnnotations with PointAnnotations, see the example here https://github.com/mapbox/mapbox-maps-android/blob/main/app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationWithPointAnnotationActivity.kt. 
At the time of pointAnnotationManager, you can specify config that takes cluster property. 

Hope this helps.

**Comment by alexviet** [comment]
I have tried your code but i don't see any method to add ViewAnnotations into point annotations. You code simply perform add a custom viewAnnotation and add a point annotation with coordinates POINT. All i want to add a lot of custom viewAnnotation as annotation then enable clustering like https://github.com/mapbox/mapbox-maps-android/blob/main/app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/PointAnnotationClusterActivity.kt#L62-L72. Sorry if i miss understood your meaning.

**Comment by boronov** [comment]
Any progress ?

**Comment by tlipov** [comment]
Any progress on this issue?

**Comment by trifagabriel** [comment]
@ank27 you made my day by redirecting me to that github example. I was researching and trying to implement clustering inside my app for two days now. Not all heroes wear caps. Thanks!!!!

**Comment by michal-bogucki** [comment]
Have you managed to add clustering with custom points?
